### ISSUE TITLE
Encrypted connections for IMAP/POP/ Webserver

### ISSUE THREAD

**Issue by Udera** [vexim/vexim2#70]
We made some efforts to implement a more secure password hashing, it would be a pity to transmit all these passwords through unencrypted connections. We have already fixed this in exim (https://github.com/avleen/vexim2/commit/4effffa57f28fd502caafce40d76f19f3df38020).

For the web-interface, we should perhaps mention to use SSL connections and point to some guides.

For IMAP/POP servers, we should perhaps provide more detailed information. It's not really difficult, I can contribute my dovecot implementation in the next days. Would be nice if someone on courier could contribute his settings too.

Qpopper?

**Comment by rimas-kudelis** [comment]
The Dovecot manual that I wrote is deliberately short, because I wanted to only provide Vexim specifics in it, not a full-fledged config.
Same goes for other readme files – in my opinion, the shorter they are, the better. Surely, mentioning which configuration options to change to have a better secured server won't hurt, but I just don't think we should resort to providing a bunch of "prepared for use" configuration files, because we don't have enough helping hands to maintain them (for one, I'm still not sure that our `configure` file works ☺).
By the way, I think we should move all these articles to the wiki once we have it bootstrapped with some content (ping @Ikonia!). Perhaps then we'll find a way to share full configs with our specifics conveniently highlighted, to please everyone. What do you think?

**Comment by Udera** [comment]
You have this problem all the time with manuals. But even if they are outdated, they provide the basic steps to follow. I have even a very old debian/vexim-guide with all basic steps (like to put the antivirus-user into the Debian-exim-group. For someone who is new to this topic, just installing SSL everywhere might be a little bit too much and he doesn't know where to start.
I'm just testing the `configure` file and it seems to work. But I don't know if it does exactly the same like the `conf.d`-files

**Comment by Z3po** [comment]
I've written a puppet manifest for doing all the installation for using vexim.
I'll cleanup what i did and push that to a github repository.
That's probably the easiest Installation-procedure we could provide?
I'm a bit confused about what vexim is. From my point of view it's a web-interface for managing my exim users/domains/aliases. Therefore we can provide configuration examples and tell them to use SSL in their connections but if the users like to use courier or dovecot or whatever they should ask their specific documentation about how to enable it....or not?

**Comment by rimas-kudelis** [comment]
@Z3po I personally see Vexim as a web-interface for managing e-mail users/domains/aliases. So far, it has been pretty server-agnostic, so the server does not necessarily have to be Exim. Yes, we haven't provided config files for any other MDA's yet, so it has been quite focused on Exim in particular until now, but I don't think it has to be.

@Udera I decided not to wait until @ikonia starts working on our wiki, and began playing with it myself. I started with the Dovecot manual, which [I've added here](https://github.com/vexim/vexim2/wiki/Server-configuration:-Dovecot). It has not really been tested (as you know, Dovecot v.2 uses slightly different config than v.1, and my original manual was written for v.1), but I think it should work. You (and anyone else) are welcome to test and improve it. Later we can decide whether or not it stays in the Wiki.

**Comment by Udera** [comment]
I added a small section for SSL certificates for dovecot v.2. Dovecot-wiki is quite good except for the SSL ciphers where I found some good settings at bettercrypto.org.

**Comment by rimas-kudelis** [comment]
Great!

**Comment by Udera** [comment]
I started to copy&paste the courier-imap: https://github.com/vexim/vexim2/wiki/Server-Configuration:-Courier-IMAP

**Comment by rimas-kudelis** [comment]
Quoting @Udera:
> I started to copy&paste the courier-imap: https://github.com/vexim/vexim2/wiki/Server-Configuration:-Courier-IMAP

I've just updated and renamed your doc. The new URL is I started to copy&paste the courier-imap: https://github.com/vexim/vexim2/wiki/Server-Configuration:-Courier (no `-IMAP` at the end). It seems this document hadn't been revised since forever, and was partially outdated, so I updated it to be more in line with my own production config.

It would be great if somebody could validate it by trying to set up Courier according to this manual and adding the bits that might be missing. Perhaps quite unexpectedly, this manual could even become the empirical proof of our product's independence from Exim, as Courier is apparently an MTA as well.
